internal/db/collection.go

```diff
@@ -711,6 +711,29 @@ func (c *collection) save(
 		doc.SetHead(link.Cid)
 	})
 
+	if c.def.Description.IsBranchable {
+		collectionCRDT := merklecrdt.NewMerkleCollection(
+			txn,
+			keys.NewCollectionSchemaVersionKey(c.Schema().VersionID, c.ID()),
+			keys.NewHeadstoreColKey(c.def.Description.RootID),
+		)
+
+		link, headNode, err := collectionCRDT.Save(ctx, []coreblock.DAGLink{{Link: link}})
+		if err != nil {
+			return err
+		}
+
+		updateEvent := event.Update{
+			Cid:        link.Cid,
+			SchemaRoot: c.Schema().Root,
+			Block:      headNode,
+		}
+
+		txn.OnSuccess(func() {
+			c.db.events.Publish(event.NewMessage(event.UpdateName, updateEvent))
+		})
```

Comment:
> Can't the receiver of the even simpli check on this side if the collection is branchable and handle correspondingly?

I hadn't considered that, I'll think about that and get back to you :)

> if we need to send another event, then I would say it serves a different purpose and in this case we should introduce another event type.

I disagree with this, it is an update event, an update of a collection.  It needs to be handled for exactly the same reasons that an update (doc) event is currently handled, and in pretty much the same way (some rework required as P2P is very focused on docIDs etc atm as noted in PR description).

Reply:
> I hadn't considered that, I'll think about that and get back to you

I think I prefer the extra events.  It feels more correct to send independent updates for the doc and the collection, especially so as the event bus is public and can be subscribed to by users.

Reply:
I also prefer the extra event but the reason for why I prefer it is missing from this implementation.

I would expect a `CreateMany` to generate only one collection block linking to all the docs created (same for updates or mix of creates and updates). This means that we would have one update event for every doc and one update event at the end for the collection.

Reply:
> I would expect a CreateMany to generate only one collection block linking to all the docs created (same for updates or mix of creates and updates). This means that we would have one update event for every doc and one update event at the end for the collection.

I was planning on building that here, but then I realised it is not specific to collection-level commits.   The same system can (should) be applied to all crdt/blocks.

Reply:
> The same system can (should) be applied to all crdt/blocks.

Can you expand on this? Why would that apply to composites or field level CRDTs?

Reply:
Just as multiple collection blocks can be created in a single transaction, so too can multiple composite and field level blocks.  Grouping any of these together into a single block should work consistently regardless of type.

E.g. if in a single transaction `Age` on `Document1` in `branchable collection 1` is updated twice, the same mechanic for rolling the would-be-two collection level commits into a single collection commit should also apply to the field and composite level commits.  

There is nothing special about collection-level commits here, although the need for it is possibly higher.

Reply:
I would say this is definitely not the case for field level CRDTs. They only point to their previous head (they don't have links to other items) plus they don't cause network updates on their own. It *may* apply to composites for the multiple updates to the same document in one transaction but I'm not sure there is much benefit to it as it would create multiple heads for the updated fields in question only to save a composite block.

Reply:
> I would say this is definitely not the case for field level CRDTs.

We disagree, and it makes me very glad I did not try and do that work within this PR :)